explorer/interpreter/type_checker.cpp

```diff
@@ -2689,1319 +2690,1347 @@ auto TypeChecker::CheckAddrMeAccess(
 auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
                                const ImplScope& impl_scope)
     -> ErrorOr<Success> {
-  if (trace_stream_->is_enabled()) {
-    *trace_stream_ << "checking " << e->kind() << " " << *e;
-    *trace_stream_ << "\n";
-  }
-  if (e->is_type_checked()) {
+  return RunWithStackSpace<ErrorOr<Success>>([&]() -> ErrorOr<Success> {
```

Comment:
It'd be nice to avoid this extra indentation. I think it's worth adding an extra `Impl` function to keep the indentation down.

Reply:
Done.